util/template/replace.go

```diff
@@ -15,8 +15,11 @@ func Replace(s string, replaceMap map[string]string, allowUnresolved bool) (stri
 	if err != nil {
 		return "", err
 	}
-
-	replacedString, err := t.Replace(replaceMap, allowUnresolved)
+	interReplaceMap := make(map[string]interface{})
```

Comment:
I did try this at first, it requires changing the code in quite a lot of places, with arguably no benefit for the generality given the current expressions only operate on strings everywhere else.

Given that we plan to change expressions/scoping in the future, I didn't want to pollute this PR with unnecessary code.

Reply:
Expressions will always be strings -- since we're operating on YAML, they're only valid in strings. That's a hard limitation unless I'm missing something

We do have a few hacks to accept `IntOrString` in places that are normally integers (and which can be very confusing). There isn't really a way to workaround that in the schema otherwise though, since at submission time it is indeed a YAML string for an expression.